[RFC] boards: msp432p401r: Add GPIO Support

DhruvaG2000 opened the pull request:
Adds GPIO Support for the TI MSP432P4012R MCU

Signed-off-by: Dhruva Gole <goledhruva@gmail.com>

DhruvaG2000 committed:
boards: msp432p401r: Add GPIO Support

Adds GPIO Support for the TI MSP432P4012R MCU

Signed-off-by: Dhruva Gole <goledhruva@gmail.com>

DhruvaG2000 commented:
This PR Doesn't build properly yet, need help on what changes need to be added exactly for adding gpio support to this board other than the gpio node in dts.

vaishnavachath commented on dts/arm/ti/msp432p4xx.dtsi:
```diff
@@ -1,6 +1,9 @@
+#include <zephyr/dt-bindings/gpio/gpio.h>
+
+#define EXC_GPIOA0  0  /* (INT_GPIOA0 - 16) = (16-16) */
 
```
@DhruvaG2000 where do you use this define?

vaishnavachath commented on dts/arm/ti/msp432p4xx.dtsi:
```diff
@@ -35,6 +38,13 @@
 		};
+		gpioa: gpio@40005000 {
+			compatible = "ti,msp432p4xx-gpio";
+			reg = <0x40005000 0x1000>;
```
Does the GPIO driver exist for this platform? I couldn't find one, where did you get this compatible from?
you might need to add something around https://github.com/zephyrproject-rtos/hal_ti/blob/master/simplelink/source/ti/devices/msp432p4xx/driverlib/gpio.c

DhruvaG2000 replied:
Thanks for your inputs vaishnav, I have started writing a GPIO driver based on your suggestions.

vaishnavachath reviewed (changes requested):
@DhruvaG2000 refer drivers/gpio/gpio_cc13xx_cc26xx.c and https://github.com/zephyrproject-rtos/hal_ti/blob/master/simplelink/source/ti/devices/cc13x2_cc26x2/driverlib/gpio.c then you will get an idea of what needs to be done to enable GPIO support for this platform.

DhruvaG2000 committed:
drivers: gpio: add msp432p4xx driver

This will start an effort to enable gpio functionality
in the msp432p4xx boards in zephyr

Signed-off-by: Dhruva Gole <goledhruva@gmail.com>

DhruvaG2000 committed:
drivers: gpio: part 2 msp432p4xx driver

The code builds ok at this point, although
I am sceptical if it will work atall.

Signed-off-by: Dhruva Gole <goledhruva@gmail.com>

DhruvaG2000 committed:
drivers: gpio: msp432: add configs and yaml

These maybe needed to actually locate and use the msp432p gpio driver

Signed-off-by: Dhruva Gole <goledhruva@gmail.com>

DhruvaG2000 committed:
drivers: gpio: msp432p4xx: remove unused define

removed unused define

Signed-off-by: Dhruva Gole <goledhruva@gmail.com>

DhruvaG2000 commented on drivers/gpio/gpio_msp432p4xx.c:
```diff
@@ -0,0 +1,237 @@
+
+#include "gpio_utils.h"
+
+static const uint8_t pinTable[] = {
```
@vaishnavachath  can you help me with where these values are supposed to come from? What do they mean?
